### ISSUE TITLE
Not Able to Use Build-Windows Guide for WSL 

### ISSUE THREAD

**Issue by molxyz** [bitcoin/bitcoin#9356]
Hello, 

This issue is about the windows guide for WSL:  https://github.com/bitcoin/bitcoin/blob/master/doc/build-windows.md

I have tried several times by following this guide but it fails to build any version.  Even with a fresh clean install of WSL this guide doesn't work.  After *make HOST=x86_64-w64-mingw32 -j4*,  it says *checking whether we are cross compiling... no*.   Then if I went on with the configure line I would get this result: 
>checking for Berkeley DB C++ headers... no
>configure: error: libdb_cxx headers missing, Bitcoin Core requires this library for wallet functionality (--disable-wallet to disable wallet functionality)

Michagogo gave us on IRC his short guide:   https://www.irccloud.com/pastebin/W0iBQZbU/  which does work in building master and 0.13.1  versions but I can't get it to build v0.13.0.  

Any help for this issue is greatly appreciated.  Thank you.

**Comment by achow101** [comment]
Did you install the general build dependencies with
`sudo apt-get install build-essential libtool autotools-dev automake pkg-config bsdmainutils curl`?

**Comment by laanwj** [comment]
I think you must have missed a step, it is not finding the dependencies that you built in the first step (in depends). This includes berkeleydb. Are you configuring bitcoin core itself using either:
```
CONFIG_SITE=$PWD/depends/x86_64-w64-mingw32/share/config.site
```
or
```
./configure --prefix=$PWD/depends/x86_64-w64-mingw32
```
Either should work, but if you don't point it at the depends on some way it won't find them.

**Comment by molxyz** [comment]
I took note while following the guide, and this is my note:   https://0bin.net/paste/6SnwTiY5pb9Pfomr#Fv8IEhJnHK12V6ptY-P5/8rZdyGrW0C85hL39t/OnU9

I crossed check every line and tested this guide over and over several times.

**Comment by laanwj** [comment]
So did I and for me it just works.

**Comment by molxyz** [comment]
I found the answer to my issue but there's something I don't understand. 
At the start of this ubuntu system, my screen looks like this: 
```C:\Users\moli>bash
molxyz@LAPTOP-O22LRA5I:/mnt/c/Users/moli$```
Then I would start typing in the commands from the guide.  What's puzzling is it works with Michagogo's guide but doesn't work with the github guide.  Today I looked at it and thought hm maybe there's something wrong with that directory.  So I typed "cd" and it went to this directory:  
```molxyz@LAPTOP-O22LRA5I:~$```
From then on everything works!  I tested and got master, 0.13.0, and 0.13.1 versions built with no problem.  Could not get v0.12.0 and v0.12.1 built though, but I will tinker on that later.  
I'm new to linux and this sub system, I thought  `molxyz@LAPTOP-O22LRA5I:/mnt/c/Users/moli$`  and  `molxyz@LAPTOP-O22LRA5I:~$` should be the same directory but I guess not?  
@achow101 and @laanwj , thank you for your help and fast replies.

**Comment by laanwj** [comment]
> I'm new to linux and this sub system, I thought molxyz@LAPTOP-O22LRA5I:/mnt/c/Users/moli$ and molxyz@LAPTOP-O22LRA5I:~$ should be the same directory but I guess not?

One is your windows "home directory" the other your Linux home directory. They are separate directories on different file systems. Your windows drive is mounted under `/mnt/c` in the Linux root. The Linux root is not directly accessible from windows (this is why there is an install step to install the executables to the windows drive).

I don't understand why it matters for the build instructions, where you put the root, though. But I don't have time to look into this right now.

**Comment by meatwas** [comment]
May be it will help someone:

I had the issue with error  configure: error: libdb_cxx headers missing

My problem was related to the step
sudo update-alternatives --config x86_64-w64-mingw32-g++ # Set the default mingw32 g++ compiler option to posix.

During first installation, I did everything like in instructions, but for some reason i didn't have choice to input a number to select appropriate mingw (posix). May be I just didn't see it. I've seen this comment "# Set the default mingw32 g++ compiler option to posix." but thought that may be this command will do everything needed. 

But during second clean installation I payed more attention and selected appropriate one with posix and then stem with make HOST=x86_64-w64-mingw32 took much much more time and everything become ok and I didn't have this error.

It's regarding official installation build-windows with WSL ubuntu 16
